tensorboard/plugins/scalar/scalars_plugin.py

```diff
@@ -95,29 +96,36 @@ def index_impl(self, ctx, experiment=None):
                 }
         return result
 
-    def scalars_impl(self, ctx, tag, run, experiment, output_format):
+    def scalars_impl(self, ctx, tag, runs, experiment, output_format):
         """Result of the form `(body, mime_type)`."""
         all_scalars = self._data_provider.read_scalars(
             ctx,
             experiment_id=experiment,
             plugin_name=metadata.PLUGIN_NAME,
             downsample=self._downsample_to,
-            run_tag_filter=provider.RunTagFilter(runs=[run], tags=[tag]),
+            run_tag_filter=provider.RunTagFilter(runs=runs, tags=[tag]),
         )
-        scalars = all_scalars.get(run, {}).get(tag, None)
-        if scalars is None:
-            raise errors.NotFoundError(
-                "No scalar data for run=%r, tag=%r" % (run, tag)
-            )
-        values = [(x.wall_time, x.step, x.value) for x in scalars]
+        result = {}
+        for run in all_scalars:
+            scalars = all_scalars.get(run, {}).get(tag, [])
+            # Note we do not raise an error if data for a given run was not
+            # found; we just return an empty array in that case.
```

Comment:
FWIW, minor clarification that #3781 only introduced a new error for the bare multiplexer path, which has subsequently been removed by #3801.

So the change here seems mostly orthogonal to the consistency concern that led me to introduce that other error; we might still argue it's nice to be able to distinguish 404 from empty array here but I don't think there's a consistency concern.

Reply:
Yes, good point; thanks. There’s definitely not a consistency concern
anymore, I just still think that it’s a good idea. (By the reference
I mostly meant to point out that in your consistency fix you chose to
standardize on the error rather than the silently empty result, but my
wording was ambiguous; sorry.)

Reply:
The comment here that we return an empty array for not-found data is not
correct with the multiplexer provider:

```
$ curl localhost:6006/data/plugin/scalars/scalarsmulti -F tag=xent/xent_1 -F runs='["lr_1E-03,conv=1,fc=2"]'
{"lr_1E-03,conv=1,fc=2": [[1563406405.3048136, 2000, 0.01962822675704956]]}
$ curl localhost:6006/data/plugin/scalars/scalarsmulti -F tag=xent/xent_1 -F runs='["lr_1E-03,conv=1,fc=2","nope"]'
{"lr_1E-03,conv=1,fc=2": [[1563406405.3048136, 2000, 0.01962822675704956]]}
```

This is because the multiplexer provider follows the relevant portion of
the data provider contract for `read_scalars`:

> The result will only contain keys for run-tag combinations that
> actually exist, which may not include all entries in the
> `run_tag_filter`.

For what it’s worth, I think that this is the sensible behavior. It’s
fine not to fail the request, but returning an empty array when we have
the opportunity to just omit the key is just introducing error.